### PR TITLE
🏷️ #535: make zod error enum

### PR DESCRIPTION
**Pull request by samrichca**
## Description of Changes

remake of https://github.com/OHCRN/platform/pull/548

### Types
- Add an enum with the Zod custom errors
- Update types and tests

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [x] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

**Review by anncatton — COMMENTED** [review]
Looks good. I would like to type the `message` arg in `addZodCustomError` to `ZodCustomError` so that any future refinements are forced to match the error dictionary.
Also, would be more readable if the enums are accessed via destructuring:
```const { ...enumNames } = ZodCustomError.enum;```

File: packages/types/src/services/consentUi/utils/zodUtils.ts [hunk 1]
```diff
@@ -19,13 +19,15 @@
...
+import { ZodCustomError } from './ZodCustomErrors.js';
+
 /**
  * Add a custom issue/error to the Zod refinement context, programmatically rather than relying on Zod schemas.
  * @param ctx Zod refinement context
  * @param message Error message. In UI use, this should be a translation key.
  * @param path Name of the field where the error will be displayed
  */
-export const addZodCustomError = (ctx: z.RefinementCtx, path: string, message: string) => {
+export const addZodCustomError = (ctx: z.RefinementCtx, path: string, message: ZodCustomError) => {
```
Comment (anncatton) [inline packages/types/src/services/consentUi/utils/zodUtils.ts:30]: 👍

File: packages/types/src/services/consentUi/utils/guardianFields.ts [hunk 2]
```diff
@@ -33,7 +33,22 @@ import {
 	requiresGuardianInformation,
 } from '../../../common/index.js';
 
-import { ZodCustomError, addZodCustomError } from './index.js';
+import { ZodCustomError } from './ZodCustomErrors.js';
```
Comment (anncatton) [inline packages/types/src/services/consentUi/utils/guardianFields.ts:36]: is the import not working coming from `./index`?

**Review by anncatton — COMMENTED** [review]
Just wondering about that changed import, otherwise looks good

File: packages/types/src/services/consentUi/utils/guardianFields.ts (hunk 2 above)
Comment (samrichca) [reply packages/types/src/services/consentUi/utils/guardianFields.ts:36]: test libraries don't like circular imports https://stackoverflow.com/questions/54310718/undefined-typescript-enum-at-compile-time

**Review by anncatton — APPROVED** [review]

**Pull request merged**